crates/pop-contracts/src/call.rs

```diff
@@ -51,24 +52,43 @@ pub struct CallOpts {
 /// * `call_opts` - options for the `call` command.
 pub async fn set_up_call(
 	call_opts: CallOpts,
-) -> anyhow::Result<CallExec<DefaultConfig, DefaultEnvironment, Keypair>> {
+) -> Result<CallExec<DefaultConfig, DefaultEnvironment, Keypair>, Error> {
 	let token_metadata = TokenMetadata::query::<DefaultConfig>(&call_opts.url).await?;
-	let manifest_path = get_manifest_path(call_opts.path.as_deref())?;
 	let signer = create_signer(&call_opts.suri)?;
 
-	let extrinsic_opts = ExtrinsicOptsBuilder::new(signer)
-		.manifest_path(Some(manifest_path))
-		.url(call_opts.url.clone())
-		.done();
+	let extrinsic_opts = match &call_opts.path {
+		// If path is a file construct the ExtrinsicOptsBuilder from the file.
+		Some(path) if path.is_file() => {
+			let artifacts = ContractArtifacts::from_manifest_or_file(None, Some(path))?;
+			ExtrinsicOptsBuilder::new(signer)
+				.file(Some(artifacts.artifact_path()))
+				.url(call_opts.url.clone())
+				.done()
+		},
+		_ => {
+			let manifest_path = get_manifest_path(call_opts.path.as_deref())?;
+			ExtrinsicOptsBuilder::new(signer)
+				.manifest_path(Some(manifest_path))
+				.url(call_opts.url.clone())
+				.done()
+		},
+	};
 
 	let value: BalanceVariant<<DefaultEnvironment as Environment>::Balance> =
 		parse_balance(&call_opts.value)?;
 
 	let contract: <DefaultConfig as Config>::AccountId = parse_account(&call_opts.contract)?;
+	// Process the argument values input by the user.
```

Comment:
```suggestion
	// Process the provided argument values.
```
From the perspective of a library crate, there is no guarantee that it has been input by a user. It could be read from a file as an example.